src/erasure.rs

```diff
@@ -183,48 +183,41 @@ pub fn generate_coding(
 
     for i in consumed..consumed + num_blobs {
         if (i % NUM_DATA) == (NUM_DATA - 1) {
-            let mut data_blobs = Vec::with_capacity(NUM_DATA);
-            let mut data_locks = Vec::with_capacity(NUM_DATA);
-            let mut data_ptrs: Vec<&[u8]> = Vec::with_capacity(NUM_DATA);
-
             info!(
                 "generate_coding start: {} end: {} consumed: {} num_blobs: {}",
                 block_start,
                 block_start + NUM_DATA,
                 consumed,
                 num_blobs
             );
+
+            let mut data_blobs = Vec::with_capacity(NUM_DATA);
+            let mut max_data_size = 0;
+
             for i in block_start..block_start + NUM_DATA {
                 let n = i % window.len();
                 trace!("window[{}] = {:?}", n, window[n].data);
                 if window[n].data.is_none() {
                     trace!("data block is null @ {}", n);
                     return Ok(());
                 }
+                let data = window[n].data.clone().unwrap();
```

Comment:
Can you avoid the clone somehow? Maybe:
`if let Some(data) = &window[n].data`

Reply:
I changed it to
```
                let data = window[n].data.clone().unwrap();
                max_data_size = cmp::max(data.read().unwrap().meta.size, max_data_size);

                data_blobs.push(data);